### ISSUE TITLE
Setting of "verify_ssl" to false doesn't work

### ISSUE THREAD

**Issue by BalazsBago** [pypa/pipenv#4596]
I get "certificate verify failed "the following error when I try to install a package with disabled "verify_ssl".

However I can install packages with directly using pip by setting the trusted hosts:
pip install --trusted-host pypi.org --trusted-host files.pytonhosted.org pyzmq
 
My machine is behinde a company firewall and usually skipping of verification solves this kind of issues, like in the case of pip.

### Expected result

A successful package installation.

### Actual result

The whole log:
 pipenv --verbose install pyzmq
Installing pyzmq...
Installing package: pyzmq
Writing supplied requirement line to temporary file: 'pyzmq'
Installing 'pyzmq'
⠸ Installing pyzmq...$ ['/home/user/.local/share/virtualenvs/user-yJKnOelZ/bin/pip', 'install', '--verbose', '--upgrade', '--exists-action=i', '-r', '/tmp/pipenv-y5p6srj0-requirements/pipenv-euacaxy4-requirement.txt', '-i', 'https://pypi.org/simple', '--trusted-host', 'pypi.org']
Using source directory: '/home/user/.local/share/virtualenvs/user-yJKnOelZ/src'
Adding pyzmq to Pipfile's [packages]...
✔ Installation Succeeded 
Pipfile.lock (ca72e7) out of date, updating to (c0f4b3)...
Locking [dev-packages] dependencies...
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
⠴ Locking...
                          ROUND 1                           
✘ Locking Failed! 
Current constraints:
  pyzmq (from -r /tmp/pipenv2zbkcm28requirements/pipenv-kxl64kqf-constraints.txt (line 3))

Finding the best candidates:
  found candidate pyzmq==21.0.1 (constraint was <any>)

Finding secondary dependencies:
  pyzmq==21.0.1             requires -
------------------------------------------------------------
Result of round 1: stable, done

Generating hashes:
  pyzmq
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 976, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connection.py", line 370, in connect
    ssl_context=context,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/util/ssl_.py", line 377, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 725, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/util/retry.py", line 439, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /pypi/pyzmq/json (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 763, in <module>
    main()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 759, in main
    parsed.requirements_dir, parsed.packages, parse_only=parsed.parse_only)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 741, in _main
    resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 709, in resolve_packages
    requirements_dir=requirements_dir,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 692, in resolve
    req_dir=requirements_dir
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1386, in resolve_deps
    req_dir=req_dir,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1096, in actually_resolve_deps
    results = resolver.clean_results()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1002, in clean_results
    collected_hashes = self.collect_hashes(ireq)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 885, in collect_hashes
    r = session.get(pkg_url, timeout=10)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /pypi/pyzmq/json (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),))
ROUND 1                           
Current constraints:
  pyzmq (from -r /tmp/pipenv2zbkcm28requirements/pipenv-kxl64kqf-constraints.txt (line 3))

Finding the best candidates:
  found candidate pyzmq==21.0.1 (constraint was <any>)

Finding secondary dependencies:
  pyzmq==21.0.1             requires -
------------------------------------------------------------
Result of round 1: stable, done

Generating hashes:
  pyzmq
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 677, in urlopen
    chunked=chunked,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 381, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 976, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connection.py", line 370, in connect
    ssl_context=context,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/util/ssl_.py", line 377, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/lib/python3.6/ssl.py", line 407, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib/python3.6/ssl.py", line 817, in __init__
    self.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 1077, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.6/ssl.py", line 689, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/connectionpool.py", line 725, in urlopen
    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/urllib3/util/retry.py", line 439, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /pypi/pyzmq/json (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 763, in <module>
    main()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 759, in main
    parsed.requirements_dir, parsed.packages, parse_only=parsed.parse_only)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 741, in _main
    resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 709, in resolve_packages
    requirements_dir=requirements_dir,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/resolver.py", line 692, in resolve
    req_dir=requirements_dir
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1386, in resolve_deps
    req_dir=req_dir,
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1096, in actually_resolve_deps
    results = resolver.clean_results()
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 1002, in clean_results
    collected_hashes = self.collect_hashes(ireq)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/utils.py", line 885, in collect_hashes
    r = session.get(pkg_url, timeout=10)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pipenv/vendor/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /pypi/pyzmq/json (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),))

### Steps to replicate

Provide the steps to replicate (which usually at least includes the commands and the Pipfile).

Command:
pipenv install pyzmq

The content of my Pipfile:
[[source]]
url = "https://pypi.org/simple"
verify_ssl = false
name = "pypi"

[packages]
pyzmq = "*"

[dev-packages]

[requires]
python_version = "3.6"

**Comment by Mause** [comment]
Which one have you tried? `ssl_verify` or `verify_ssl`? You mention both in your post

**Comment by BalazsBago** [comment]
> Which one have you tried? `ssl_verify` or `verify_ssl`? You mention both in your post

Sorry for the confusion, I only tried 'verify_ssl'.

**Comment by matteius** [comment]
@BalazsBago I believe this is resolved with `2022.1.8`  Please try it out.

```
matteius@matteius-VirtualBox:~/shared-projects/pipenv-triage/pipenv-4596$ pipenv --verbose install pyzmq
Using python: 3.10
Path to python: /home/matteius/.pyenv/versions/3.10.2/bin/python3
Creating a virtualenv for this project...
Pipfile: /home/matteius/shared-projects/pipenv-triage/pipenv-4596/Pipfile
Using /home/matteius/.pyenv/versions/3.10.2/bin/python3 (3.10.2) to create virtualenv...
⠇ Creating virtual environment...created virtual environment CPython3.10.2.final.0-64 in 268ms
  creator CPython3Posix(dest=/home/matteius/.virtualenvs/pipenv-4596-y0gaapBq, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/matteius/.local/share/virtualenv)
    added seed packages: pip==22.0.3, setuptools==60.9.3, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment! 
Virtualenv location: /home/matteius/.virtualenvs/pipenv-4596-y0gaapBq
Installing pyzmq...
Installing package: pyzmq
Writing supplied requirement line to temporary file: 'pyzmq'
Installing 'pyzmq'
$ /home/matteius/.virtualenvs/pipenv-4596-y0gaapBq/bin/python -m pip install --verbose --upgrade --exists-action=i -r /tmp/pipenv-_o52475n-requirements/pipenv-mq50qfax-requirement.txt -i https://pypi.org/simple
Using source directory: '/home/matteius/.virtualenvs/pipenv-4596-y0gaapBq/src'
Adding pyzmq to Pipfile's [packages]...
✔ Installation Succeeded 
Pipfile.lock not found, creating...
Locking [dev-packages] dependencies...
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
Reporter.starting()
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.starting()
Reporter.adding_requirement(SpecifierRequirement('pyzmq'), None)
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.adding_requirement(SpecifierRequirement('pyzmq'), None)
Reporter.starting_round(0)
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.starting_round(0)
Reporter.pinning(LinkCandidate('https://files.pythonhosted.org/packages/e6/3d/f0a39a4e94b4db31561a3431d035ddf53b52fc8c0dc09cfc704b24d58f0f/pyzmq-22.3.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl#sha256=ea12133df25e3a6918718fbb9a510c6ee5d3fdd5a346320421aac3882f4feeea (from https://pypi.org/simple/pyzmq/) (requires-python:>=3.6)'))
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.pinning(LinkCandidate('https://files.pythonhosted.org/packages/e6/3d/f0a39a4e94b4db31561a3431d035ddf53b52fc8c0dc09cfc704b24d58f0f/pyzmq-22.3.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl#sha256=ea12133df25e3a6918718fbb9a510c6ee5d3fdd5a346320421aac3882f4feeea (from https://pypi.org/simple/pyzmq/) (requires-python:>=3.6)'))
Reporter.ending_round(0, state)
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.ending_round(0, state)
Reporter.starting_round(1)
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.starting_round(1)
Reporter.ending(State(mapping=OrderedDict([('pyzmq', LinkCandidate('https://files.pythonhosted.org/packages/e6/3d/f0a39a4e94b4db31561a3431d035ddf53b52fc8c0dc09cfc704b24d58f0f/pyzmq-22.3.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl#sha256=ea12133df25e3a6918718fbb9a510c6ee5d3fdd5a346320421aac3882f4feeea (from https://pypi.org/simple/pyzmq/) (requires-python:>=3.6)'))]), criteria={'pyzmq': Criterion((SpecifierRequirement('pyzmq'), via=None))}))
INFO:pipenv.patched.notpip._internal.resolution.resolvelib.reporter:Reporter.ending(State(mapping=OrderedDict([('pyzmq', LinkCandidate('https://files.pythonhosted.org/packages/e6/3d/f0a39a4e94b4db31561a3431d035ddf53b52fc8c0dc09cfc704b24d58f0f/pyzmq-22.3.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl#sha256=ea12133df25e3a6918718fbb9a510c6ee5d3fdd5a346320421aac3882f4feeea (from https://pypi.org/simple/pyzmq/) (requires-python:>=3.6)'))]), criteria={'pyzmq': Criterion((SpecifierRequirement('pyzmq'), via=None))}))
⠙ Locking...
✔ Success! 
Updated Pipfile.lock (e3ebda)!
Installing dependencies from Pipfile.lock (e3ebda)...
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 0/0 — 00:00:00
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
```

**Comment by BalazsBago** [comment]
It already works with pipenv 2021.5.29.
I made a test with a local devpi server.
